pkg/reconciler/labeler/v2/accessors.go

```diff
@@ -182,31 +197,53 @@ func NewConfigurationAccessor(
 }
 
 // list implements Accessor
-func (c *Configuration) list(ns, name string) ([]kmeta.Accessor, error) {
-	cl, err := c.configurationLister.Configurations(ns).List(labels.SelectorFromSet(labels.Set{
-		serving.RouteLabelKey: name,
-	}))
+func (c *Configuration) list(ns, routeName string, state v1.RoutingState) ([]kmeta.Accessor, error) {
+	cl, err := c.configurationLister.Configurations(ns).List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 	// Need a copy to change types in Go
-	kl := make([]kmeta.Accessor, len(cl))
-	for i, c := range cl {
-		kl[i] = c
+	kl := make([]kmeta.Accessor, 0, len(cl))
+	for _, c := range cl {
+		if containsListAnnValue(c.Annotations, serving.RoutesAnnotationKey, routeName) {
+			kl = append(kl, c)
+		}
 	}
 	return kl, err
 }
 
+// containsListAnnValue returns true if value is present in a comma-separated annotation.
+func containsListAnnValue(annotations map[string]string, key, value string) (has bool) {
+	_, has = getListAnnValue(annotations, key, value)
+	return
+}
```

Comment:
```suggestion
// containsListAnnValue returns true if value is present in a comma-separated annotation.
func containsListAnnValue(annotations map[string]string, key, value string) bool {
	_, has := getListAnnValue(annotations, key, value)
	return has
}
```
It's obviously a personal preference, but I think we mostly do this in knative, with empty return feeling a bit unnatural.